insights/parsers/xfs_info.py

```diff
@@ -20,60 +102,14 @@ class XFSInfo(Parser):
-            the report, keyed on the five section names in the output:
-            '``meta-data``', '``data``', '``naming``', '``log``', and
-            '``realtime``'.  '``meta-data``', '``data``' and '``log``' are
-            always present.  Within each dictionary a special key
-            '``specifier``' stores any data immediately after the section
-            name - e.g. '/dev/sda' or 'version' in the case of the output
-            below.  Any data immediately following that is stored in the
-            ``specifier_value`` key.  Otherwise, data is read in key=value
-            pairs - e.g. from the output below, the ``isize`` key will have
-            the value ``32`` (an integer).  Data values given in blocks are
-            left as is, so the value of the ``agsize`` key is '16777184 blks'
-            as a string.
-        mount(str): If the mount point can be derived from the file name of
-            the original output, then this attribute contains the
-            reconstructed mount point name.
-        device(str): The device name immediately after the '``meta-data``'
-            section heading.
-        data_size(int): The size of the data segment in bytes, from
-            multiplying the ``blocks`` and ``bsize`` values of the ``data``
-            section together.
-        log_size(int): The size of the log segment in bytes, from multiplying
-            the ``blocks`` and ``bsize`` values of the ``log`` section
-            together.
-
-    Sample output (from file '``sos_commands/xfs/xfs_info_.data``')::
-
-         meta-data=/dev/sda      isize=256    agcount=32, agsize=16777184 blks
-                  =              sectsz=512   attr=2
-         data     =              bsize=4096   blocks=536869888, imaxpct=5
-                  =              sunit=32     swidth=128 blks
-         naming   =version 2     bsize=4096
-         log      =internal      bsize=4096   blocks=32768, version=2
-                  =              sectsz=512   sunit=32 blks, lazy-count=1
-         realtime =none          extsz=524288 blocks=0, rtextents=0
-
-    Examples:
-        >>> xfs_list = shared[XFSInfo] # A list of devices
-        >>> for xfs in xfs_list:
-        ...     print 'Mount {mnt} -> {dev}: {blks} blocks, {bytes} bytes'.format(
-        ...         mnt=xfs.mount, dev=xfs.device,
-        ...         blks=xfs.xfs_info['data']['blocks'], bytes=xfs.data_size
-        ...     )
-        Mount /data -> /dev/sda: 536869888 blocks, 2199019061248 bytes
-        >>> data = xfs_list[0].xfs_info  # Direct access to data dictionary
-        >>> data['meta-data']['specifier']
-        '/dev/sda'
-        >>> data['meta-data']['agsize']  # blocks values not parsed
-        '16777184 blks'
-        >>> data['realtime']['rtextents']
-        0
     """
 
+    def __init__(self, context):
+        self.xfs_info = {}
+        self.data_size = 0
+        self.log_size = 0
```

Comment:
How about shifting the initialization of `self.mount` and `self.device` from the parse_content to here too?

Reply:
A good point, might as well.